pkg/rbac/userData.go

```diff
@@ -277,10 +277,82 @@ func (user *UserDataCache) updateUserManagedClusterList(cache *Cache, ns string)
 	}
 }
 
+// Request the SelfSubjectRullesRreview(SSRR) for the namespace and process the rules.
+// TO-DO-Separate-PR: Reduce the args required by this func.
+func (user *UserDataCache) getSSRRforNamespace(ctx context.Context, cache *Cache, ns string,
+	userInfo authv1.UserInfo, wg *sync.WaitGroup, lock *sync.Mutex) {
+	defer wg.Done()
+
+	// Request the SelfSubjectRulesReview for the namespace.
+	rulesCheck := authz.SelfSubjectRulesReview{
+		Spec: authz.SelfSubjectRulesReviewSpec{
+			Namespace: ns,
+		},
+	}
+	result, err := user.getAuthzClient().SelfSubjectRulesReviews().Create(ctx, &rulesCheck, metav1.CreateOptions{})
+	if err != nil {
+		klog.Error("Error creating SelfSubjectRulesReviews for namespace", err, ns)
+	} else {
+		klog.V(9).Infof("SelfSubjectRulesReviews Kube API result for ns:%s : %v\n", ns, prettyPrint(result.Status))
+	}
+
+	// Process the SSRR result and add to this UserDataCache object.
+	for _, rules := range result.Status.ResourceRules {
+		for _, verb := range rules.Verbs {
+			if verb == "list" || verb == "*" {
+				for _, res := range rules.Resources {
+					for _, api := range rules.APIGroups {
+						lock.Lock()
```

Comment:
Wondering would it be better to lock after we find resource with verb == "list" || verb ="*" (third line in loop)?

Reply:
Maybe. I don't have a definite answer but here are my thoughts on the pros and cons.
- Here we lock for the shortest period of time. This creates a large number of lock/unlock cycles which allows other go routines to get the lock and continue their processing.
- Moving to the outside loop we have less lock/unlock cycles, which requires less computation, but the hold the lock for a longer period blocking other go routines.

Reply:
these are good points. we can revisit later as potential optimization.